### PR TITLE
Add a WebSocket Client example. As Per issue #465

### PR DESCRIPTION
**Pull request by tigerpixel**
Add a WebSocket Client example class to match the WebSocket server example.

### Motivation:

To provide an example of the usage of the WebSocket client upgrade functionality.
To satisfy issue #465

### Modifications:

Added a ‘WebSocketClient’ example with a readme and a target in the package file.

### Result:

There is now a WebSocket client example to assist with the creation of a WebSocket client.

### CODE REVIEW

**Comment by weissi** [comment]
@swift-nio-bot please test

**Comment by weissi** [comment]
@swift-nio-bot test this please

File: Sources/NIOWebSocketClient/main.swift [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        let frame = WebSocketFrame(fin: true, opcode: .connectionClose, data: data)
+        context.write(self.wrapOutboundOut(frame)).whenComplete { (_: Result<Void, Error>) in
+            context.close(mode: .output, promise: nil)
+        }
+    }
+}
+
+extension ByteBuffer {
+    fileprivate init(string: String) {
+        self = ByteBufferAllocator().buffer(capacity: string.utf8.count)
```
Comment (Lukasa) [inline Sources/NIOWebSocketClient/main.swift]: As these are intended to be examples of proper NIO usage, can we use the `ByteBufferAllocator` from the `Channel`?

File: Sources/NIOWebSocketClient/main.swift [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    }
+    
+    private func pingTestFrameData(context: ChannelHandlerContext) {
+        let frameData = ByteBuffer(string: self.testFrameData)
+        let frame = WebSocketFrame(fin: true, opcode: .ping, data: frameData)
+        context.write(self.wrapOutboundOut(frame), promise: nil)
+    }
+    
+    private func pong(context: ChannelHandlerContext, frame: WebSocketFrame) {
+        if let frameData = frame.data.getString(at: 0, length: self.testFrameData.count) {
```
Comment (Lukasa) [inline Sources/NIOWebSocketClient/main.swift]: This may as well be `readString`, which will save you from the incorrect index usage you have here.

Comment (Lukasa) [reply Sources/NIOWebSocketClient/main.swift]: (To call that just take a local mutable copy of the `WebSocketFrame`).

**Comment by tigerpixel** [comment]
Apologies for the delay. Both done.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Pull request merged**